### ISSUE TITLE
SubCommand Pattern Matching fails if a subcommand is prefix of another subcommand using cmd interface

### ISSUE THREAD

**Issue by KedarisettiSreeVamsi** [gofr-dev/gofr#686]
**Describe the bug**
Let's take an example where there are 2 subcommands available here I am taking it as 'h' and 'hello' where h refers to help command and hello for hello world response as in the provided examples.

**To Reproduce**
Steps to reproduce the behavior, if applicable:

1. The code is

   ```go
      package main

      import "gofr.dev/pkg/gofr"

      func main(){

      app := gofr.NewCmd()
     
      app.SubCommand("h",func(c *gofr.Context) (interface{},error){
              return "help",nil
      })

      app.SubCommand("hello",func(c *gofr.Context) (interface{},error){
              return "Hello World!",nil
      })
      
      app.Run()
     }
   ```

2. The issue is when we run 

  
![image](https://github.com/gofr-dev/gofr/assets/34485436/6b6b684e-bab6-43ee-9dc6-5fbaa1de4272)

**Expected behavior**

It should actually attempt to run Hello World but ran h Subcommand providing the output as help.

**Screenshots**

**Environments (please complete the following information):**
- OS: [e.g. Linux]
- gofr version [e.g. v1.5.0]
- go version [e.g. 1.21]

**More description**

**Comment by KedarisettiSreeVamsi** [comment]
Please assign this issue to me as I have already provided a fix to it via #675 .

**Comment by KedarisettiSreeVamsi** [comment]
@aryanmehrotra @srijan-27 @vipul-rawat can someone assign this issue to me.
